selectrum.el

```diff
@@ -717,16 +722,23 @@ just rendering it to the screen and then checking."
           (if (or (and highlighted-index
                        (< highlighted-index 0))
                   (and (not selectrum--match-required-p)
-                       (not displayed-candidates)))
+                       (not displayed-candidates))
+                  (and selectrum--default-candidate
+                        (not (member selectrum--default-candidate
+                                     selectrum--refined-candidates))))
               (if (= (minibuffer-prompt-end) bound)
                   (let ((str
                          (propertize
                           (format " [default value: %S]"
-                                  (or selectrum--default-candidate 'none))
+                                  (or (and selectrum--default-candidate
+                                           (substring-no-properties
+                                            selectrum--default-candidate))
+                                      'none))
                           'face 'minibuffer-prompt))
                         (ol (make-overlay
-                             (minibuffer-prompt-end)
-                             (minibuffer-prompt-end))))
+                             ;; Put cursor after overlay.
+                             (1- (minibuffer-prompt-end))
+                             (1- (minibuffer-prompt-end)))))
```

Comment:
What happens if this change was omitted? I thought that we already handled all the edge cases of cursor positioning w.r.t. overlays, but I'd also be willing to believe we missed one.

Reply:
Without this change, the default candidate is written after the cursor. It doesn't change how anything works, but I thought it was better to keep the cursor after the entire prompt message.